extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -83,12 +90,24 @@ export class PlatformService implements IPlatformService {
 		this._outputChannel.show(preserveFocus);
 	}
 
-	osType(platform: string = this.platform()): OsType {
+	osType(): OsType {
+		if (!this._osType) {
+			throw new Error('platformService was not initialized');
+		}
+		return this._osType;
+	}
+	private async getOsType(platform: string = this.platform()): Promise<OsType> {
 		if (Object.values(OsType).includes(<OsType>platform)) {
```

Comment:
Why do you have this platform param? It's never used.

Reply:
(and to be clear - it's never used I mean that the only place this is called doesn't pass one in so you always use the default this.platform)

Reply:
Removed the parameter.